server/automute_preferences.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+func (p *Plugin) PreferencesHaveChanged(c *plugin.Context, preferences []model.Preference) {
+	p.updateAutomutingOnPreferencesChanged(c, preferences)
+}
+
+func (p *Plugin) updateAutomutingOnPreferencesChanged(c *plugin.Context, preferences []model.Preference) {
+	userIDsToEnable, userIDsToDisable := getUsersWhoChangedPlatform(preferences)
+
+	for _, userID := range userIDsToEnable {
+		if connected, err := p.isUserConnected(userID); err != nil {
+			p.API.LogError(
+				"Unable to potentially enable automute for user",
+				"UserID", userID,
+				"Error", err.Error(),
+			)
+		} else if !connected {
+			continue
+		}
```

Comment:
I'm wondering if we should remove this check: users who haven't connected will still get "spammed" if the counterparty in the DM is connected, and it might be desirable to effect this setting manually. Thoughts, @johndavidlugtu?

Reply:
I think that would probably work as expected then since an unconnected MM-only user will still get notifications for their DMs. The only way they'd get extra notifications is if they're on Teams and MM without being connected which I can't see us supporting

Reply:
Referencing the other PR where I left a comment to keep one thread: https://github.com/mattermost/mattermost-plugin-msteams-sync/pull/479#issuecomment-1947350958